### PR TITLE
683 join keys@main

### PR DESCRIPTION
**Pull request by nikolas-burkoff**
Part of https://github.com/insightsengineering/teal/issues/683

### CODE REVIEW

File: tests/testthat/test-FilteredData.R [hunk 1]
```diff
@@ -24,44 +24,29 @@ testthat::test_that("get_keys returns the same character array if a dataset has
...
 })
 
-testthat::test_that("get_join_keys returns empty character if no join_keys", {
+testthat::test_that("get_join_keys returns NULL if no join_keys", {
   filtered_data <- FilteredData$new(list(iris = list(dataset = head(iris), keys = "test")), join_keys = NULL)
-  testthat::expect_equal(filtered_data$get_join_keys(), character(0))
+  testthat::expect_null(filtered_data$get_join_keys())
 })
 
-testthat::test_that("get_join_keys returns empty character if no join_keys for specific datasets", {
```
Comment (mhallal1) [inline tests/testthat/test-FilteredData.R:32]: do we need this test somewhere?
```
filtered_data <- FilteredData$new(
  list(
    iris = list(dataset = head(iris)),
    iris2 = list(dataset = head(iris)),
    iris3 = list(dataset = head(iris))
  ),
  join_keys = teal.data::join_keys(
    teal.data::join_key("iris2", "iris3", c("Species" = "Species"))
  )
)

rr <- filtered_data$get_join_keys()
rr$get("iris", "iris2")
> character(0)

```

Comment (nikolas-burkoff) [reply tests/testthat/test-FilteredData.R:32]: So I don't think we need this any more as we have tested `JoinKeys$get()` in `teal.data` and in teal.slice we have tested `FilteredData$get_join_keys()` so no need to check:

```
x <-  FilteredData$get_join_keys()
x$get(y,z)
```
works

Comment (nikolas-burkoff) [reply tests/testthat/test-FilteredData.R:32]: previously we had the function `FilteredData$get_join_keys(y, z)` which needed testing, especially as it had some logic in it

Comment (mhallal1) [reply tests/testthat/test-FilteredData.R:32]: fair enough.

**Review by mhallal1 — APPROVED** [review]

**Pull request merged**